pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/realtime/segment/FlushThresholdUpdaterTest.java

```diff
@@ -18,699 +18,354 @@
  */
 package org.apache.pinot.controller.helix.core.realtime.segment;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Random;
-import org.apache.pinot.common.config.TableConfig;
+import java.util.Arrays;
+import org.apache.pinot.common.config.TableNameBuilder;
 import org.apache.pinot.common.metadata.segment.LLCRealtimeSegmentZKMetadata;
-import org.apache.pinot.common.partition.PartitionAssignment;
-import org.apache.pinot.common.utils.CommonConstants;
 import org.apache.pinot.common.utils.LLCSegmentName;
-import org.apache.pinot.core.realtime.impl.fakestream.FakeStreamConfigUtils;
 import org.apache.pinot.core.realtime.stream.PartitionLevelStreamConfig;
 import org.apache.pinot.core.realtime.stream.StreamConfig;
-import org.apache.pinot.core.realtime.stream.StreamConfigProperties;
-import org.testng.Assert;
-import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
-/*
- * TODO: FIXME
- * - Consider changing the test to use mock streamConfig than the one generated via real configurations.
- *   Makes things a lot easier. Or, have a class that inherits from StreamConfig and has setters/getters
- *   to return the values that we care about.
- * - Change the SegmentSizeBasedFlushThresholdUpdater class to use a method to get time, and fake it
```

Comment:
not sure if this TODO has been implementated. if so, then please change the TODO to the test to use fake time instead (this PR is already too big)

Reply:
Added the TODO